### PR TITLE
Fix remaining accessibility issues

### PR DESCRIPTION
**Pull request by TimboTambo**
# Fix remaining accessibility issues

Relates to #925 #926 #924 #922

### Acceptance Criteria

- [ ] _The first thing this PR must achieve_
- [ ] _The second thing this PR must achieve_
- [ ] _etc_

### Testing information

_notes that might be helpful for testing_

### Pre-flight Checklist

- [ ] Testing aligns with [testing strategy][testing strategy]
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/web/src/pages/product/index.tsx [hunk 1]
```diff
@@ -172,6 +172,7 @@ const App: NextPage = () => {
   return (
     <Page
       title="Products"
+      metaTitle={productName ? `Products | ${productName}` : 'Products'}
```
Comment (TimboTambo) [inline medicines/web/src/pages/product/index.tsx:175]: Replace with "Product results" or "Search results"

File: medicines/web/src/pages/substance-index/index.tsx [hunk 2]
```diff
@@ -51,6 +51,7 @@ const App: NextPage = () => {
   return (
     <Page
       title="Products"
+      metaTitle={substanceIndex ? `Products | ${substanceIndex}` : 'Products'}
```
Comment (TimboTambo) [inline medicines/web/src/pages/substance-index/index.tsx:54]: Substance index

File: medicines/web/src/pages/substance/index.tsx [hunk 3]
```diff
@@ -68,6 +68,7 @@ const App: NextPage = () => {
   return (
     <Page
       title="Products"
+      metaTitle={substanceName ? `Products | ${substanceName}` : 'Products'}
```
Comment (TimboTambo) [inline medicines/web/src/pages/substance/index.tsx:71]: Substance

**Review by StuartHarris — APPROVED** [review]
Amazing! And a lot of work :-) Well done.

**Review by StuartHarris — APPROVED** [review]
Amazing! And a lot of work :-) Well done.

**Pull request merged**